cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -548,4 +551,53 @@ private MDSKey getNamespaceKey(@Nullable String name) {
     }
     return builder.build();
   }
+
+  public void recordWorkflowProgramStart(Id.Program program, String programRunId, String workflow,
+                                         String workflowRunId, String workflowNodeId, long startTimeInSeconds,
+                                         String twillRunId) {
+    // Get the run record of the Workflow which started this program
+    MDSKey key = new MDSKey.Builder()
+      .add(TYPE_RUN_RECORD_STARTED)
+      .add(program.getNamespaceId())
+      .add(program.getApplicationId())
+      .add(ProgramType.WORKFLOW.name())
+      .add(workflow)
+      .add(workflowRunId)
+      .build();
+
+    RunRecord record = get(key, RunRecord.class);
+    if (record == null) {
+      String msg = String.format("No meta found for associated Workflow %s run record %s, while recording run for the" +
+                                   " namespace %s app %s type %s program %s runid %s", workflow, workflowRunId,
+                                 program.getNamespaceId(), program.getApplicationId(), program.getType().name(),
+                                 program.getId(), programRunId);
+      LOG.error(msg);
+      throw new IllegalArgumentException(msg);
+    }
+
+    deleteAll(key);
```

Comment:
Ohh I see what you are saying. Yes, we are not changing the key also, so we do not need `deleteAll`. Removed in the last commit.